src/rules/noUnboundMethodRule.ts

```diff
@@ -150,3 +225,15 @@ function isSafeUse(node: ts.Node): boolean {
             return false;
     }
 }
+
+function isWhitelisted(node: ts.Node, whitelist: Set<string>, allowTypeof: boolean): boolean {
+    if (isTypeOfExpression(node.parent)) {
+        return allowTypeof;
+    }
+    if (isCallExpression(node.parent)) {
+        const expression = node.parent.expression as ts.Identifier;
```

Comment:
> `as ts.Identifier`

It's actually possible for `node.parent.expression` to not be an identifier. You'll want to add test cases for:

```ts
(condition ? expectA : expectB)(c.method);
(await someObject)(c.method);
(await someMethod())(c.method);
```

I don't think we should worry about trying to allow these even if `expectA` and `expectB` are whitelisted. Suggestion: just don't run this check if the parent's expression isn't an identifier.

Reply:
Ok, changed as you suggested. I maybe might include it in some future PR